kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -49,15 +52,38 @@ def _hash(string: str) -> str:
     return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
 
 
-def _get_hashed_username():
+def _get_or_create_uuid():
+    """
+    Reads a UUID from a configuration file or generates and saves a new one if not present.
+    """
+    config_path = user_config_dir("kedro")
+    full_path = os.path.join(config_path, CONFIG_FILENAME)
+    config = ConfigParser()
+
     try:
-        username = getpass.getuser()
-        return _hash(username)
-    except Exception as exc:
-        logger.warning(
-            "Something went wrong with getting the username. Exception: %s",
-            exc,
-        )
+        if os.path.exists(full_path):
+            config.read(full_path)
+
+            if config.has_section("telemetry") and "uuid" in config["telemetry"]:
+                try:
+                    return uuid.UUID(config["telemetry"]["uuid"]).hex
+                except ValueError:
+                    pass  # Invalid UUID found, will generate a new one
```

Comment:
> If the file does exist but it's empty, or if the id is not a valid UUID4, or if the file cannot be written, or if anything else fails, then the identity in _send_heap_event must not be set (None or empty string).

The only situation in which we should generate a new UUID is if the file doesn't exist

Reply:
@astrojuanlu , could you please clarify why it's preferable to always send a Null userId when encountering a broken UUID, rather than attempting to fix it by creating and saving a new ID once?

Reply:
Because this leaves a way for the user to overwrite the UUID with something invalid or empty so it's not sent

Reply:
(as a way to opt out)